torch/csrc/autograd/variable.cpp

```diff
@@ -139,6 +142,7 @@ const std::shared_ptr<Node>& Variable::grad_fn() const {
     if (!diff_view_meta->grad_fn_ && !diff_view_meta->base_.requires_grad()) {
       return diff_view_meta->grad_fn_;
     }
+#ifndef C10_MOBILE
```

Comment:
as mentioned over chat, this part seems difficult to maintain.

The rest doesn't seem to be adding that much complexity, because we already expose AutoNonVariableType as part of the pseudo-public interface and you are just slicing along that border.

Reply:
The reason this is commented out is, I assume, because of the direct reference to `generated::AsStridedBackward`? That doesn't seem too bad to me.

Reply:
Yes, generated::AsStridedBackward is the main blocker.
Ok, I updated the PR to keep the Functions.h/cpp for now to keep generated::AsStridedBackward and remove #if/#endif here.
In next PR I can update the codegen script to only generate this single function for mobile build - Does it sound better than the gating? @gchanan